python/chronos/src/bigdl/chronos/data/tsdataset.py

```diff
@@ -812,3 +816,41 @@ def _check_basic_invariants(self, strict_check=False):
         # check dt sorted
         if strict_check:
             _check_dt_is_sorted(self.df, self.dt_col)
+
+    def get_best_lookback(self, aggregate='mode', top_k=3):
+        """
+        Use the calculation of Fourier transform and autocorrelation coefficient to
+        get the time period of all columns, and select the appropriate training time
+        period for the model by specifying aggregate.
```

Comment:
We don't need to talk too much detail here. We just put "calculate the cycle length of the time series in this TSDataset".